File: Examples/erlang/hello.erl
# Erlang Hello World
# Time Warp Studio - Erlang example

-module(hello).
-export([main/0, greet/1, factorial/1]).

main() ->
    io:format("Hello, World!~n"),
    io:format("Erlang: concurrent, functional, fault-tolerant.~n"),

    % Greeting
    Name = "Time Warp",
    greet(Name),

    % Factorials
    io:format("~nFactorials:~n"),
    lists:foreach(
        fun(N) ->
            F = factorial(N),
            io:format("  ~w! = ~w~n", [N, F])
        end,
        [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
    ),

    % List operations
    Numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    Evens = lists:filter(fun(X) -> X rem 2 == 0 end, Numbers),
    Squares = lists:map(fun(X) -> X * X end, Numbers),
    Sum = lists:sum(Numbers),

    io:format("~nNumbers: ~w~n", [Numbers]),
    io:format("Evens: ~w~n", [Evens]),
    io:format("Squares: ~w~n", [Squares]),
    io:format("Sum: ~w~n", [Sum]).

greet(Name) ->
    io:format("Welcome to ~s Studio!~n", [Name]).

factorial(0) -> 1;
factorial(N) when N > 0 -> N * factorial(N - 1).

File: Examples/rust/hello.rs
// Rust Hello World
// Time Warp Studio - Rust example

fn main() {
    println!("Hello, World!");
    println!("Rust: systems programming with memory safety.");

    // Variables and types
    let name = "Time Warp";
    let version: i32 = 10;
    println!("Welcome to {} Studio v{}!", name, version);

    // Arithmetic
    let sum: i32 = (1..=10).sum();
    println!("Sum of 1 to 10: {}", sum);

    // Vectors
    let languages = vec!["Rust", "Ruby", "Erlang", "Python", "Lua"];
    println!("\nLanguages:");
    for lang in &languages {
        println!("  - {}", lang);
    }

    // if/else
    let x = 42;
    if x > 100 {
        println!("\n{} is large", x);
    } else if x > 10 {
        println!("\n{} is medium", x);
    } else {
        println!("\n{} is small", x);
    }

    // match
    let day = 3;
    let day_name = match day {
        1 => "Monday",
        2 => "Tuesday",
        3 => "Wednesday",
        4 => "Thursday",
        5 => "Friday",
        _ => "Weekend",
    };
    println!("Day {}: {}", day, day_name);

    // Loop with accumulator
    let mut total = 0;
    for i in 1..=5 {
        total += i * i;
    }
    println!("\nSum of squares 1..5: {}", total);
}
